src/plugins/intel_npu/src/al/include/intel_npu/al/config/config.hpp

```diff
@@ -373,6 +415,25 @@ void OptionsDesc::add() {
     }
 }
 
+template <class Opt>
+void OptionsDesc::add(compilerVersion compilerVersionRequirement) {
+    if (compilerVersionRequirement.vclMajor >= Opt::compilerSupportVersion().vclMajor &&
+        compilerVersionRequirement.vclMinor >= Opt::compilerSupportVersion().vclMinor) {
+        OPENVINO_ASSERT(_impl.count(Opt::key().data()) == 0, "Option '", Opt::key().data(), "' was already registered");
+        _impl.insert({Opt::key().data(), details::makeOptionModel<Opt>()});
+
+        for (const auto& deprecatedKey : Opt::deprecatedKeys()) {
+            OPENVINO_ASSERT(_deprecated.count(deprecatedKey.data()) == 0,
+                            "Option '",
+                            deprecatedKey.data(),
+                            "' was already registered");
+            _deprecated.insert({deprecatedKey.data(), Opt::key().data()});
+        }
+    } else {
+        // Option is being skipped > log a warning here
```

Comment:
Silent if an option is dropped or not supported, please add log here. By the way, log level is warning for developer build and error for release version,  which kind of log level shall we use here to notice user about the issue?

Reply:
Yes, will do. Adding logging is till on the todo. I believe a warning would be enough for this one. The user should always query supported_properties list first, and only operate on properties from that list, this is our official standing. If they set an unsupported property it will result in a throw telling the user that the property is not supported. A log here will serve mostly just debug purposes i believe.